### DISCUSSION TITLE
is it possible to add data-* attributes to select <option>?

### DISCUSSION THREAD

**Discussion by okliv** [jeremyevans/forme#92] [General]
hi!

i'm trying to combine forme+[select2](https://select2.org
)+[select2-to-tree](https://github.com/clivezhg/select2-to-tree)

select2-to-tree [needs](https://github.com/clivezhg/select2-to-tree#2-directly-create-the-select-elementsee-example-2-in-exampleexamplehtml) `data-pup`, `class` and `non-leaf` attributes for each option

```html
<option value="11" data-pup="1" class="l2 non-leaf">opt_11</option>
```

is it possible to collect somehow `:options` for `f.select` with that attributes?

i tried to use generic `f.tag`

```ruby
f.tag(:div, class: 'form-group') do
  f.tag(:label, { for: input.name }, input.label)
  f.tag(input.type, **attrs) do # here data: {a: 'b'} renders as data="{:a=>'b'}" instead of data-a="b" as it works for f.select
    input.options.each do |o|
    f.tag(:option, { value: o.value, 'data-pup' => o.parent, class: o.css_class }, [o.title])
  end
end
```

but it doesn't work nice with [`:data`](https://github.com/jeremyevans/forme#general-options-), `:add_blank` options and other magic `f.select` has

**Comment by jeremyevans** [comment]
You can do this by making the second entry in the array for the options be a hash:

```ruby
puts Forme.form{|f| f.input(:select, :options=>[["text", {"value"=>1, "data-pup"=>"foo", "class"=>"bar", "non-leaf"=>"baz"}]])}.to_s
# <form><select><option class="bar" data-pup="foo" non-leaf="baz" value="1">text</option></select></form>
```

This was unfortunately not documented in the README.  I'll fix that shortly.

**Reply by jeremyevans** [reply]
Fixed at 08f9672b63c099ef382de4ffc305e891d27fc69e

**Reply by okliv** [reply]
hi! almost same problem with `input`.

now i can't add arbitrary attribute to it

``` ruby
f.input type: select, id: ..., name: ..., foo: 'baz'
```
`foo="baz"` is absent in rendered html. it simply ignores it. what am i doing wrong?

(i use [unpoly](https://unpoly.com/), it is based on custom html attributes up-*)

thanks

**Reply by jeremyevans** [reply]
To use arbitrary attributes, you need to use the `:attr` option.
